### ISSUE TITLE
ErrorPolicy: Ignore and All not changing the response

### ISSUE THREAD

**Issue by VijoPlays** [apollographql/apollo-client#10339]
**Intended outcome:**

Defining an Error Policy should behave according to the documentation (e.g. "all" should return an error and partial data, etc).

(Another thing I realized: The documentation refers to apollo-link - though that has been merged into the client repro and thus that part is no longer needed.)

**Actual outcome:**
I have 2 things I retrieve from GraphQL. A is a working endpoint, B is broken (to test error handling). When calling just A, we get data. Calling A and B together returns just an error (so far everything behaves correctly). 
Setting an errorPolicy (see below) results in a response similar to not setting any Policy at all: I only receive the errors.

**How to reproduce the issue:**

Simply set errorPolicy in a request:

```
let { loading, error, data } = useQuery<Data>(GET_DATA, {
    variables: {
      name: name,
    },
    errorPolicy: "all"
  });
```

The errorPolicy makes no difference here.

I also tried replicating the request in Postman similar to:

```
{"operationName":"...","errorPolicy":"ignore", "variables":{...},"query":"..."}
```

With no luck - the parameter is also just being ignored.

**Versions**

I ran the command, but it didn't copy anything to my clipboard. The versions I run on my system:

Node: 16.17.1
apollo-client: latest (3.7.1, I believe)
React: ^18.0.8

**Comment by bignimbus** [comment]
Hi @VijoPlays 👋🏻  thanks for filing this issue!  I'm sure the community can help you figure out what might be going on.  I appreciate the description you provided but unfortunately it might be tough to reproduce this issue without understanding the whole application.  Can you provide a [runnable reproduction](https://github.com/apollographql/react-apollo-error-template/generate) to facilitate a resolution to your issue?  Thanks so much 🙏🏻

**Comment by bignimbus** [comment]
@VijoPlays can you clarify whether you think the issue is on the server side or the client side?  After rereading I'm not 100% certain I understand the expected vs. actual behavior.

**Comment by VijoPlays** [comment]
Right, I tinkered around with it a little more and I found where my problem lied:

> In my Go code (gqlgen) I wrote a method that will first log a graphQL error and then throw an error - though the GraphQL server doesn't like that, it seems. Yesterday I tinkered around with removing the errors from it, but because the errors were no longer thrown I missed a part of the code that probably threw another error.

Fix for that is to simply not throw errors (instead only use 'graphql.AddError(ctx, errors.New("some err"))'), probably work with a bool right now - but I'll check with the doc for that project/file a bug over there.

---

Anywho, sorry for wasting your time and thanks for the help!

@bignimbus as for the [Apollo-Link reference](https://www.apollographql.com/docs/react/data/error-handling/#advanced-error-handling-with-apollo-link) I'd personally add a small note at the beginning of the page that it used to be a separate library, but is now part of the client (if I got that right). It's not a big deal, but I just stumbled over it and found it strange when I searched for Link. If that's the case I also wouldn't mind filing a PR later. 🙂
